### PR TITLE
PR: Move updates workers to workers module

### LINKED ISSUE #9025: Move update workers to utils and integrate to worker manager framework
- [ ] Move `spyder/workers` to `spyder/utils/workers` and integrate to worker managers in `spyder/utils/workers`.
- [ ] Move `spyder/utils/workers.py` to `spyder/utils/workers/manager.py`

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

- Unify the worker usage using the worker manager
- Reorganize and simplify code
- Added new tests for worker manager
- Updated tests for spyder update checking utils

### Issue(s) Resolved

Fixes #9025

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

**Comment by ccordoba12** [comment]
Tests are still failing on Linux. And why did remove the WorkerManager tests in your last commit?

**Comment by ccordoba12** [comment]
Tests are also generating a segfault on Windows and pip (Windows/Python 3.7 is tested with pip in Azure).

**Comment by goanpeca** [comment]
I am not sure what is going on on Linux and the segfault on pip and windows seems ubrelated to this as I have seen it on other PRs.

The test for the workers seem to be causing issues so i removed to see how i can make them work.

**Comment by ccordoba12** [comment]
> I am not sure what is going on on Linux

You need to install a VM and see what's happening there.

> the segfault on pip and windows seems ubrelated to this as I have seen it on other PRs.

Yes, but the failure here is happening after like 5 tests and it's very predictable, so it's clearly introduced by this PR.

**Comment by goanpeca** [comment]
@ccordoba12 you know why the circle ci checks are not appearing?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # https://dev.azure.com/spyder-ide/spyder/
 
+variables:
+  RUNNING_AZURE: true  # this is available to all jobs
```
Comment (ccordoba12) [inline azure-pipelines.yml]: There's an `AZURE` variable defined below. Please use that instead of defining a new one.

Comment (goanpeca) [reply azure-pipelines.yml]: Ok

**Pull request closed without merging**